Linter fails after adding storybook config files

<!-- Please do your best to fill out all of the sections below! -->

## Current Behavior
<!-- What is the behavior that currently you experience? -->
Linting a library which has storybook configurations causes lint errors. 

## Expected Behavior
<!-- What is the behavior that you expect to happen? -->
<!-- Is this a regression? .i.e Did this used to be the behavior at one point?  -->
Linter should success without errors.

## Steps to Reproduce
<!-- Help us help you by making it easy for us to reproduce your issue! -->

<!-- Can you reproduce this on https://github.com/nrwl/nx-examples? -->
<!-- If so, open a PR with your changes and link it below. -->
<!-- If not, please provide a minimal Github repo -->
<!-- At the very least, provide as much detail as possible to help us reproduce the issue -->

<!-- Remove this line -->
1. Create a workspace
```
npx create-nx-workspace workspace

? What to create in the new workspace react
? Application name                    web
? Default stylesheet format           emotion
? Use Nx Cloud? (It's free and doesn't require registration.) No
Creating a sandbox with Nx...
```

2. Add a library
```
npm run nx g lib components
```

3. Add Storybook configurations
```
npm i -D @nrwl/storybook
npm run nx g storybook-configuration components

? Configure a cypress e2e app to run against the storybook instance? Yes
? Automatically generate story files for components declared in this library? Yes
? Automatically generate *.spec.ts files in the cypress e2e app generated by the cypress-configure schematic? Yes
```

4. Run lint
```
npm run lint components
```


### Failure Logs
<!-- Please include any relevant log snippets or files here. -->
```
> nx workspace-lint && nx lint "components"


> nx run components:lint 

Linting "components"...

/Users/puku0x/Desktop/workspace/libs/components/.storybook/main.js
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: libs/components/.storybook/main.js.
The file must be included in at least one of the projects provided

/Users/puku0x/Desktop/workspace/libs/components/.storybook/preview.js
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: libs/components/.storybook/preview.js.
The file must be included in at least one of the projects provided

/Users/puku0x/Desktop/workspace/libs/components/.storybook/webpack.config.js
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: libs/components/.storybook/webpack.config.js.
The file must be included in at least one of the projects provided

✖ 3 problems (3 errors, 0 warnings)

Lint errors found in the listed files.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! workspace@0.0.0 lint: `nx workspace-lint && nx lint "components"`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the workspace@0.0.0 lint script.
```

### Environment
<!-- It's important for us to know the context in which you experience this behavior! -->
<!-- Please paste the result of `nx report` below! -->
```
>  NX  Report complete - copy this into the issue template

  nx : Not Found
  @nrwl/angular : Not Found
  @nrwl/cli : 10.3.0
  @nrwl/cypress : 10.3.0
  @nrwl/eslint-plugin-nx : 10.3.0
  @nrwl/express : Not Found
  @nrwl/jest : 10.3.0
  @nrwl/linter : 10.3.0
  @nrwl/nest : Not Found
  @nrwl/next : Not Found
  @nrwl/node : Not Found
  @nrwl/react : 10.3.0
  @nrwl/schematics : Not Found
  @nrwl/tao : 10.3.0
  @nrwl/web : 10.3.0
  @nrwl/workspace : 10.3.0
  typescript : 4.0.3
```

--- a/e2e/storybook/src/storybook.test.ts
+++ b/e2e/storybook/src/storybook.test.ts
@@ -27,6 +27,20 @@ describe('Storybook schematics', () => {
       ).toContain(`<title>Storybook</title>`);
     }, 1000000);
 
+    it('should lint a React based storybook without errors', () => {
+      newProject();
+
+      const reactStorybookLib = uniq('test-ui-lib-react');
+      runCLI(`generate @nrwl/react:lib ${reactStorybookLib} --no-interactive`);
+      runCLI(
+        `generate @nrwl/react:storybook-configuration ${reactStorybookLib} --generateStories --no-interactive`
+      );
+
+      // build React lib
+      const output = runCLI(`run ${reactStorybookLib}:lint`);
+      expect(output).toContain('All files pass linting.');
+    }, 1000000);
+
     it('should build a React based storybook that references another lib', () => {
       const proj = newProject();
 

--- a/packages/storybook/migrations.json
+++ b/packages/storybook/migrations.json
@@ -25,6 +25,11 @@
       "version": "11.0.12",
       "description": "Update storybook if installed and above 6",
       "factory": "./src/migrations/update-11-0-12/update-storybook"
+    },
+    "update-11-5-3": {
+      "version": "11.5.3-beta.1",
+      "description": "Update react storybook lint config",
+      "factory": "./src/migrations/update-11-5-3/update-lint-ignores"
     }
   },
   "packageJsonUpdates": {

--- a/packages/storybook/src/generators/configuration/project-files/.storybook/tsconfig.json__tmpl__
+++ b/packages/storybook/src/generators/configuration/project-files/.storybook/tsconfig.json__tmpl__
@@ -4,5 +4,5 @@
     "emitDecoratorMetadata": true
   },
   "exclude": ["../**/*.spec.ts" <% if(uiFramework === '@storybook/react') { %>, "../**/*.spec.js", "../**/*.spec.tsx", "../**/*.spec.jsx"<% } %>],
-  "include": ["../src/**/*"]
+  "include": ["../src/**/*", "*.js"]
 }

--- a/packages/storybook/src/migrations/update-11-5-3/update-lint-ignores.spec.ts
+++ b/packages/storybook/src/migrations/update-11-5-3/update-lint-ignores.spec.ts
@@ -1,0 +1,58 @@
+import { Tree } from '@angular-devkit/schematics';
+import { getFileContent } from '@nrwl/workspace/testing';
+import { runMigration } from '../../utils/testing';
+
+describe('Update 11-5-3', () => {
+  let tree: Tree;
+
+  beforeEach(async () => {
+    tree = Tree.empty();
+
+    // create workspace
+    tree.create(
+      'workspace.json',
+      JSON.stringify({
+        projects: {
+          ['home-ui-react']: {
+            projectType: 'library',
+            root: 'libs/home/ui-react',
+            sourceRoot: 'libs/home/ui-react/src',
+            architect: {
+              storybook: {
+                builder: '@nrwl/storybook:storybook',
+                options: {
+                  uiFramework: '@storybook/react',
+                  port: 4400,
+                  config: {
+                    configFolder: 'libs/home/ui-react/.storybook',
+                  },
+                },
+              },
+            },
+          },
+        },
+      })
+    );
+
+    tree.create(
+      'libs/home/ui-react/.storybook/tsconfig.json',
+      JSON.stringify({
+        extends: '../tsconfig.json',
+        include: ['../src/**/*'],
+      })
+    );
+  });
+
+  it(`should add storybook tsconfig to lint target and update tsconfigs in project for React project`, async () => {
+    tree = await runMigration('update-11-5-3', {}, tree);
+
+    expect(getFileContent(tree, 'libs/home/ui-react/.storybook/tsconfig.json'))
+      .toMatchInlineSnapshot(`
+      "{
+        \\"extends\\": \\"../tsconfig.json\\",
+        \\"include\\": [\\"../src/**/*\\", \\"./*.js\\"]
+      }
+      "
+    `);
+  });
+});

--- a/packages/storybook/src/migrations/update-11-5-3/update-lint-ignores.ts
+++ b/packages/storybook/src/migrations/update-11-5-3/update-lint-ignores.ts
@@ -1,0 +1,94 @@
+import * as path from 'path';
+import {
+  chain,
+  Tree,
+  SchematicContext,
+  Rule,
+} from '@angular-devkit/schematics';
+
+import {
+  formatFiles,
+  readJsonInTree,
+  updateWorkspaceInTree,
+  serializeJson,
+} from '@nrwl/workspace';
+
+import { isFramework, TsConfig } from '../../utils/utilities';
+import { normalize } from '@angular-devkit/core';
+
+interface ProjectDefinition {
+  root: string;
+  sourceRoot: string;
+  projectType: 'library' | 'application';
+
+  schematic?: Record<string, any>;
+  architect: Record<
+    string,
+    import('@angular-devkit/core').workspaces.TargetDefinition
+  >;
+}
+
+export default function (tree: Tree, context: SchematicContext) {
+  return chain([update, formatFiles()]);
+}
+
+function update(tree: Tree, context: SchematicContext): Rule {
+  return updateWorkspaceInTree((config, context, tree) => {
+    Object.entries<ProjectDefinition>(config.projects).forEach(
+      ([projectName, projectConfig]) => {
+        updateTsconfigInclude(tree, context, { projectName, projectConfig });
+      }
+    );
+
+    return config;
+  });
+}
+
+function updateTsconfigInclude(
+  tree: Tree,
+  context: SchematicContext,
+  options: {
+    projectName: string;
+    projectConfig: ProjectDefinition;
+  }
+) {
+  const architect = options.projectConfig.architect;
+
+  const paths = {
+    tsConfigStorybook: normalize(
+      path.join(options.projectConfig.root, '.storybook/tsconfig.json')
+    ),
+  };
+
+  const hasStorybookConfig =
+    architect.storybook && tree.exists(paths.tsConfigStorybook);
+
+  if (!hasStorybookConfig) {
+    context.logger.info(
+      `${options.projectName}: no storybook configured. skipping migration...`
+    );
+    return;
+  }
+
+  const isReactProject = isFramework('react', {
+    uiFramework: architect.storybook.options?.uiFramework as Parameters<
+      typeof isFramework
+    >[1]['uiFramework'],
+  });
+
+  const tsConfig = {
+    storybook: readJsonInTree<TsConfig>(tree, paths.tsConfigStorybook),
+  };
+
+  if (isReactProject && Array.isArray(tsConfig.storybook.include)) {
+    tsConfig.storybook.include = uniqueArray([
+      ...tsConfig.storybook.include,
+      './*.js',
+    ]);
+    tree.overwrite(paths.tsConfigStorybook, serializeJson(tsConfig.storybook));
+  }
+}
+
+function uniqueArray<T extends Array<any>>(value: T) {
+  return [...new Set(value)] as T;
+}

